src/core/CMakeLists.txt

```diff
@@ -155,19 +154,22 @@ target_include_directories(EspressoCore PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
 if(WITH_COVERAGE)
   target_compile_options(EspressoCore PUBLIC "$<$<CONFIG:Release>:-g>")
   target_compile_options(EspressoCore PUBLIC "$<$<CONFIG:Release>:-O0>")
-  target_compile_options(EspressoCore PUBLIC "$<$<CXX_COMPILER_ID:Clang>:-fprofile-instr-generate>")
-  target_compile_options(EspressoCore PUBLIC "$<$<CXX_COMPILER_ID:Clang>:-fcoverage-mapping>")
-  target_compile_options(EspressoCore PUBLIC "$<$<NOT:$<CXX_COMPILER_ID:Clang>>:--coverage>")
-  target_compile_options(EspressoCore PUBLIC "$<$<NOT:$<CXX_COMPILER_ID:Clang>>:-fprofile-arcs>")
-  target_compile_options(EspressoCore PUBLIC "$<$<NOT:$<CXX_COMPILER_ID:Clang>>:-ftest-coverage>")
-  if (NOT CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
+  target_compile_options(
+    EspressoCore PUBLIC "$<$<CXX_COMPILER_ID:Clang>:-fprofile-instr-generate>")
+  target_compile_options(
+    EspressoCore PUBLIC "$<$<CXX_COMPILER_ID:Clang>:-fcoverage-mapping>")
+  target_compile_options(EspressoCore
+                         PUBLIC "$<$<NOT:$<CXX_COMPILER_ID:Clang>>:--coverage>")
+  target_compile_options(
+    EspressoCore PUBLIC "$<$<NOT:$<CXX_COMPILER_ID:Clang>>:-fprofile-arcs>")
+  target_compile_options(
+    EspressoCore PUBLIC "$<$<NOT:$<CXX_COMPILER_ID:Clang>>:-ftest-coverage>")
+  if(NOT CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
     target_link_libraries(EspressoCore PUBLIC gcov)
   endif()
 endif()
 
-target_link_libraries(EspressoCore PUBLIC EspressoConfig utils)
-target_link_libraries(EspressoCore PUBLIC Boost::serialization Boost::mpi MPI::MPI_CXX Random123)
-target_link_libraries(EspressoCore PRIVATE Profiler)
+target_link_libraries(EspressoCore PUBLIC EspressoConfig utils Boost::serialization Boost::mpi MPI::MPI_CXX Random123 shapes PRIVATE Profiler)
```

Comment:
Do all of them have to be `PUBLIC`? E.g. iirc `Random123` is not used on the interface of core.

Reply:
I have checked...

* Random123: https://github.com/espressomd/espresso/blob/5aab7f7ec5012ee1247de667772f140720829c81/src/core/random.hpp#L29

will check the others